osclib/stagingapi.py

```diff
@@ -1072,25 +1072,27 @@ def map_ring_package_to_subject(self, project, pkg):
 
         return project
 
-    def get_sub_packages(self, pkg, project=None):
+    def get_sub_packages(self, package, project=None):
         """
         Returns a list of packages that need to be linked into rings
         too. A package is actually a tuple of project and package name
         """
         ret = []
         if not project:
-            project = self.ring_packages.get(pkg)
+            project = self.ring_packages.get(package)
         if not project:
             return ret
-        url = self.makeurl(['source', project, pkg],
+        url = self.makeurl(['source', project, package],
                            {'cmd': 'showlinked'})
 
         # showlinked is a POST for rather bizzare reasons
         f = http_POST(url)
         root = ET.parse(f).getroot()
 
         for pkg in root.findall('package'):
-            ret.append((pkg.get('project'), pkg.get('name')))
+            # ensure sub-package is valid
+            if pkg.get('project') in self.rings and pkg.get('name') != package:
```

Comment:
I am not sure what you mean. #948 was created to solve #946 created by you which asked for adi requests to support creating subpackages. So yes, we do have plans. Either way the mere fact that this would break such a feature indicates that the check is likely not correct since being in a ring is not the only "valid" case.